spec/grid_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "spec_helper"
+require_relative "../lib/grid.rb"
+
+RSpec.describe Grid do
+  describe "#coordinate_valid?" do
```

Comment:
Each of these has an inside and an outside in each test, and I am not sure that in the right way to think about the boundary. Let's make those explicit in separate tests:

`it "is true if in the last row"`
`it "is false when after the last row"`

... which makes me think, is it true of false if it is before the first row?